ic-agent/src/hash_tree.rs

```diff
@@ -0,0 +1,274 @@
+/** Hash-tree
+
+#### Resources:
+
+- [Public spec link](https://hydra.dfinity.systems/latest/dfinity-ci-build/ic-ref.pr-218/public-spec/1/index.html#_encoding_of_certificates)
```

Comment:
Can we point the URL to the published (still internal) version of the spec that implements it? Something like https://docs.dfinity.systems/public/v/0.12.2/. Using a link to a rendered PR is bad form IMO (and the PR might be updated or rejected).

Reply:
Yep, that's fine with me.